modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/autocomplete/Autocomplete.js

```diff
@@ -12,69 +12,62 @@
  * details.
  */
 
-import ClayAutocomplete from '@clayui/autocomplete';
-import ClayDropDown from '@clayui/drop-down';
-import {FocusScope} from '@clayui/shared';
-import {ReactPortal, useIsMounted} from '@liferay/frontend-js-react-web';
-import PropTypes from 'prop-types';
-import React, {useEffect, useRef, useState} from 'react';
+import {useIsMounted} from '@liferay/frontend-js-react-web';
+import {Autocomplete as BaseAutocomplete} from 'frontend-js-components-web';
+import React, {useEffect, useState} from 'react';
 
-import {debouncePromise} from '../../utilities/debounce';
+import debounce from '../../utilities/debounce';
 import {AUTOCOMPLETE_VALUE_UPDATED} from '../../utilities/eventsDefinitions';
-import {useLiferayModule} from '../../utilities/hooks';
 import {getData, getValueFromItem} from '../../utilities/index';
 import {showErrorNotification} from '../../utilities/notifications';
 import InfiniteScroller from '../infinite_scroller/InfiniteScroller';
 
-function Autocomplete({onChange, onItemsUpdated, onValueUpdated, ...props}) {
-	const [query, setQuery] = useState(props.initialLabel || '');
-	const [initialised, setInitialised] = useState(
-		Boolean(props.customViewModuleUrl || props.customView)
-	);
-	const [active, setActive] = useState(false);
-	const [selectedItem, updateSelectedItem] = useState(props.initialValue);
-	const [items, updateItems] = useState(null);
+function Autocomplete({
+	apiUrl,
+	autofill = false,
+	customView: CustomView,
+	disabled = false,
+	fetchDataDebounce = 200,
+	id,
+	infiniteScrollMode = false,
+	initialLabel = '',
+	initialValue = '',
+	inputClass,
+	inputId,
+	inputName,
+	inputPlaceholder = Liferay.Language.get('type-here'),
+	itemsKey,
+	itemsLabel,
+	onChange,
+	onValueUpdated,
+	pageSize = 10,
+	required,
+}) {
+	const [query, setQuery] = useState(initialLabel || '');
+	const [initialised, setInitialised] = useState(Boolean(CustomView));
+	const [selectedItem, setSelectedItem] = useState(initialValue);
+	const [items, setItems] = useState(null);
 	const [loading, setLoading] = useState(false);
-	const [totalCount, updateTotalCount] = useState(null);
-	const [lastPage, updateLastPage] = useState(null);
-	const [page, updatePage] = useState(1);
-	const [pageSize, updatePageSize] = useState(props.pageSize);
-	const node = useRef();
-	const dropdownNode = useRef();
-	const inputNode = useRef();
-	const FetchedCustomView = useLiferayModule(props.customViewModuleUrl);
```

Comment:
I removed this part and the `customViewModuleUrl` prop because I didn't see this being used anywhere

Reply:
That was a requirement from @marco-leo to let external developer using their own views by passing their module url. 
We do something similar [here](https://github.com/liferay/liferay-portal/blob/60c178b91f14b75e9bd4b56cf43cf152c1dbb824/modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/data_set_display/views/table/TableCell.js#L100) - I created the hook `useLiferayModule` during the migration so it's not used there, I think it can be moved out of commerce modules for better reusability. 

We don't use it internally but somebody theoretically could.

NB:
I just realised we have the same `useLiferayModule` function declared twice 🤦